### PR TITLE
Symbolab: Better expanded triggering for IA

### LINKED ISSUE #3017: Advanced Calculator: overtriggering
This IA is triggering on queries related to C++, trying to solve the equation.  An example would be "java c++ integration."

---

IA Page:  http://duck.co/ia/view/symbolab 
[Maintainer](http://docs.duckduckhack.com/maintaining/guidelines.html): @levaly

### PR DESCRIPTION
**Pull request by levaly**
## Type of Change

- [ ] New Instant Answer
    - [ ] Cheat Sheet
- [X] Improvement
    - [ ] Bug fix
    - [X] Enhancement
- [ ] Non–Instant Answer
    - [ ] Other (Role, Template, Test, Documentation, etc.)

## Description of new Instant Answer, or changes

Better triggering support
Supports more cases, like factoring, expanding, equations such as x^2+x+2=0
Handles better false cases
Over 500 test cases

## Related Issues and Discussions
Fixes #3017

## People to notify
@moollaza @edgesince84 @russellholt

---

IA Page: http://duck.co/ia/view/symbolab

Forum Topic:

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by moollaza** [comment]
@levaly thanks for the updates and the amazing test coverage. I'm deploying this to beta to test 👍

File: lib/DDG/Spice/Symbolab.pm [hunk 1]
```diff
@@ -2,17 +2,132 @@ package DDG::Spice::Symbolab;
...
+	"arccsc" => 2,
+	
+	"cosh" => 2,
+	"sinh" => 2,
+	"tanh" => 2,
+	"coth" => 2,
+	"sech" => 2,
+	"csch" => 2,
+	
+	"sqrt" => 2,
```
Comment (moollaza) [inline lib/DDG/Spice/Symbolab.pm]: perhaps"square root" as well?

**Comment by moollaza** [comment]
@levaly Thanks! This LGTM :+1:

It's live on Beta: https://beta.duckduckgo.com/?q=compute%20integral%20of%20x^2%20dx

File: lib/DDG/Spice/Symbolab.pm (hunk 1 above)
Comment (levaly) [inline lib/DDG/Spice/Symbolab.pm]: sure, I will add it shortly, and push another PR.

File: lib/DDG/Spice/Symbolab.pm [hunk 2]
```diff
@@ -2,17 +2,132 @@ package DDG::Spice::Symbolab;
...
-    if ($_ =~ /[\^\*\=\+\-\/\\]/ or $_ =~ /\b(sin|cos|tan|cot|csc|sec|ln|log|sqrt)/i) {
-        return $_;
-    }
-    return;
+    #for cases like c++, j++
+	if ($_ =~ /\+\+/){  
+	  return;
+	}
+	
+	my $minWeight = 2;
```
Comment (moollaza) [inline lib/DDG/Spice/Symbolab.pm]: This can be defined outside the `handle` so we don't recreate the variable each time the `handle` function runs 👍

Comment (levaly) [inline lib/DDG/Spice/Symbolab.pm]: ok, will change

**Comment by levaly** [comment]
I don't see the IA showing up on beta.

**Comment by moollaza** [comment]
@levaly try this? https://beta.duckduckgo.com/?q=compute+integral+of+x%5E2+dx&ia=calculator

**Comment by levaly** [comment]
thanks, the interface was a bit confusing for me :)

**Comment by levaly** [comment]
@moollaza the above mentioned changes, should they go into the next version, or are you waiting for them for this version?

**Pull request merged**